corehq/apps/accounting/utils/stripe.py

```diff
@@ -0,0 +1,51 @@
+from decimal import ROUND_DOWN, Decimal
+from django.conf import settings
+from corehq.apps.accounting.utils import log_accounting_info
+import stripe
+
+
+def get_customer_cards(username):
+    from corehq.apps.accounting.models import StripePaymentMethod, PaymentMethodType
+
+    try:
+        payment_method = StripePaymentMethod.objects.get(
+            web_user=username,
+            method_type=PaymentMethodType.STRIPE
+        )
+        stripe_customer = payment_method.customer
+        return dict(stripe_customer.cards)
+    except StripePaymentMethod.DoesNotExist:
+        pass
+    except stripe.error.AuthenticationError:
+        if not settings.STRIPE_PRIVATE_KEY:
+            log_accounting_info("Private key is not defined in settings")
+        else:
+            raise
+    return None
+
+
+def charge_through_stripe(card, customer, amount_in_dollars, currency, description, idempotency_key=None):
+    """
+    Creates a charge on a customer's card using Stripe's payment processing service.
+
+    This function is a simple wrapper around the Stripe API's `Charge.create` method.
+
+    Parameters:
+    - card (str): The card token or ID representing the payment source to be charged.
+    - customer (str): The ID of the stripe customer to whom the card belongs.
```

Comment:
Late to the party, but it looks like the way this is currently used, a customer object is passed in? I'm looking at https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/accounting/models.py#L3863-L3863 which should call into [this code](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/accounting/models.py#L3709-L3731), but I know the test for this method passes in an id.

Reply:
I think you're right.
But [stripe.Charge.create api doc](https://stripe.com/docs/api/charges/create) said customer should be the id of the customer... maybe we don't need this customer parameter at all???? Let me think about it...

Reply:
From the Stripe dashboard, I can see both customer id and the customer object will work. But in order to make our code clear and consistent, I will fix other codes that pass customer object. Thanks for flagging this!

Reply:
Though in the api doc, it stated that customer parameter is optional. But if I call the api without customer parameter, I will get: `stripe.error.InvalidRequestError: Request req_k567XiwpQHJa6n: You must provide a value for 'customer'`.

Reply:
Fixed in 6f2129c890d401daee9a003e69c4c1cc47299a4e

Reply:
6f2129c will failed two tests, which I don't know how to fix... So instead, I updated the docstring in a1413f082d762fae37ae277af7f8ddf220e48755. Because from the stripe dashboard, no matter I pass in an customer object or customer id, they know how to handle it...